test/hotspot/jtreg/compiler/intrinsics/TestIntegerDivMod.java

```diff
@@ -0,0 +1,111 @@
+import java.util.random.RandomGenerator;
+import java.util.random.RandomGeneratorFactory;
+
+
+public class TestIntegerDivMod {
+    private final int BUFFER_SIZE = 1024;
+    private int[] dividends = new int[BUFFER_SIZE];
+    private int[] divisors = new int[BUFFER_SIZE];
+    private int[] quotients =  new int[BUFFER_SIZE];
+    private int[] remainders =  new int[BUFFER_SIZE];
+    private RandomGenerator rng;
+
+    public static void main(String args[]) {
+        TestFramework.run(TestIntegerDivMod.class);
+    }
+
+    public TestIntegerDivMod() {
+        rng = RandomGeneratorFactory.getDefault().create(0);
+        for (int i = 0; i < BUFFER_SIZE; i++) {
+            dividends[i] = rng.nextInt();
+            divisors[i] = rng.nextInt();
+        }
+    }
+
+    @Test // needs to be run in (fast) debug mode
+    @Warmup(10000)
+    @IR(counts = {"UDivI", ">= 1"}) // Atleast one UDivI node is generated if intrinsic is used
+    public void testDivideUnsigned() {
+        for (int i = 0; i < BUFFER_SIZE; i++) {
+            quotients[i] = Integer.divideUnsigned(dividends[i], divisors[i]);
+        }
+        checkResult("divideUnsigned");
+    }
+
+    @Test // needs to be run in (fast) debug mode
+    @Warmup(10000)
+    @IR(counts = {"UModI", ">= 1"}) // Atleast one UModI node is generated if intrinsic is used
+    public void testRemainderUnsigned() {
+        for (int i = 0; i < BUFFER_SIZE; i++) {
+            remainders[i] = Integer.remainderUnsigned(dividends[i], divisors[i]);
+        }
+        checkResult("remainderUnsigned");
+    }
+
+
+    @Test // needs to be run in (fast) debug mode
+    @Warmup(10000)
+    @IR(counts = {"UDivModI", ">= 1"}) // Atleast one UDivModI node is generated if intrinsic is used
+    public void testDivModUnsigned() {
+        for (int i = 0; i < BUFFER_SIZE; i++)  divmod(dividends[i], divisors[i], i);
+        checkResult("divmodUnsigned");
+    }
+
+    private void divmod(int dividend, int divisor, int i) {
+        quotients[i] = Integer.divideUnsigned(dividend, divisor);
+        remainders[i] = Integer.remainderUnsigned(dividend, divisor);
+    }
+
+    public void checkResult(String mode) {
+        for (int i=0; i < BUFFER_SIZE; i++) {
+            long dividend = Integer.toUnsignedLong(dividends[i]);
+            long divisor = Integer.toUnsignedLong(divisors[i]);
+
+            int quo =  (int) (dividend / divisor);
+            int rem = (int) (dividend % divisor);
+            boolean mismatch;
+            switch (mode) {
+                case "divideUnsigned": mismatch = (quotients[i] != quo); break;
+                case "remainderUnsigned": mismatch = (remainders[i] != rem); break;
+                case "divmodUnsigned": mismatch = (quotients[i] != quo || remainders[i] != rem); break;
+                default: throw new IllegalArgumentException("incorrect mode");
+            }
+            if (mismatch) {
+                throw new RuntimeException("Test failed");
```

Comment:
It would be good to print dividend, divisor, operation, actual result and expected result here.

Reply:
Please see the updated error message in the recent commit.